### PR TITLE
Allow elasticache redis updates

### PR DESCRIPTION
**Pull request by aguilinger**
Allows update for elasticache redis given the new operation.

https://app.shortcut.com/aptible/story/9282/update-for-awselasticacheredis

### CODE REVIEW

File: internal/provider/redis/resource.go [hunk 1]
```diff
@@ -68,38 +68,52 @@ func resourceRedisRead(ctx context.Context, d *schema.ResourceData, m any) diag.
 	_ = d.Set("environment", out.EnvironmentId)
 	_ = d.Set("id", out.Id)
 	_ = d.Set("handle", out.Handle)
+	_ = d.Set("redis_version", out.RedisVersion)
+	_ = d.Set("snapshot_retention_limit", out.SnapshotRetentionLimit)
+	_ = d.Set("snapshot_window", out.SnapshotWindow)
+	_ = d.Set("preferred_maintenance_window", out.PreferredMaintenanceWindow)
 	_ = d.Set("redis_endpoint", out.RedisEndpoint)
 	_ = d.Set("redis_port", out.RedisPort)
 	_ = d.Set("authentication_token", out.AuthenticationToken)
```
Comment (madhuravius) [inline internal/provider/redis/resource.go:77]: node_type is missing on this read, not sure if intended or if this is purely supplied from state (so we could detect drift on read, right?)

File: tests/redis/update/redis_update_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+		redisTokenOriginal := terraform.Output(t, terraformOptions, "token")
+		redisPort := terraform.Output(t, terraformOptions, "port")
+
+		assert.Equal(t, "6379", redisPort)
+
+		terraformOptionsUpdateMinorVersion := terraform.WithDefaultRetryableErrors(t, &terraform.Options{
+			TerraformDir: ".",
+			Lock:         false,
+			Vars: map[string]interface{}{
+				"aptible_aws_account_id":       os.Getenv("APTIBLE_AWS_ACCOUNT_ID"),
```
Comment (madhuravius) [inline tests/redis/update/redis_update_test.go:42]: just sanity checking you tested the other vars to make sure you're good here, i saw you updated `preferred_maintenance_window` and the `snapshot_retention_limit`, but the `snapshot_window` looks the same in all your tests.

if you tested that manually and its fine via sweetness, i can't contest, lgtm

Comment (aguilinger) [reply tests/redis/update/redis_update_test.go:42]: I did test manually but the snapshot window also changes from the initial one

File: internal/provider/redis/resource.go (hunk 1 above)
Comment (aguilinger) [reply internal/provider/redis/resource.go:77]: I don't think this was actually needed I set these trying to debug the state issue. Node type is set on the api, but it gets properly updated in state. I'll update here to be consistent but this is a case where it's unclear which value it's using

**Review by madhuravius — APPROVED** [review]

File: internal/provider/redis/resource.go (hunk 1 above)
Comment (madhuravius) [reply internal/provider/redis/resource.go:78]: resolved in dms but i think that this is something to be watchful of (if end state is unpredictable) as our update in client should return the latest each time

in dms it was more about detecting state drifts on reads or starts of various flows if multiple people worked on the same resources and needed updates on client side

**Pull request merged**